libraries/botbuilder/src/botFrameworkAdapter.ts

```diff
@@ -145,6 +145,17 @@ export class BotFrameworkAdapter extends BotAdapter implements IUserTokenProvide
             this.credentials.oAuthEndpoint = GovernmentConstants.ToChannelFromBotLoginUrl;
             this.credentials.oAuthScope = GovernmentConstants.ToChannelFromBotOAuthScope;
         }
+
+        // Relocate the tenantId field used by MS Teams to a new location (from channelData to conversation)
+        // This will only occur on actities from teams that include tenant info in channelData but NOT in conversation,
+        // thus should be future friendly.  However, once the the transition is complete. we can remove this.
+        this.use(async(context, next) => {
+            if (context.activity && context.activity.conversation && !context.activity.conversation.tenantId && context.activity.channelData && context.activity.channelData.tenant) {
+                context.activity.conversation.tenantId = context.activity.channelData.tenant.id;
+            }
+            await next();
+        });
```

Comment:
Do we need this delta in C# SDK too?

Reply:
I think you should add constraint on channelId first
**context.channelId == ChannelTypes.Teams**

---
In reply to: [277836819](https://github.com/Microsoft/botbuilder-js/pull/886#discussion_r277836819) [](ancestors = 277836819)

Reply:
@tomlm - @gabog has done the C# version of thi.

i'll add the extra constraint now!

Reply:
@tomlm - there is no ChannelTypes object in the JS SDK -- the closest I could find is part of the prompt definitions in dialogs.  I did a test against channelId === 'msteams'